### PR TITLE
mailhog 0.1.8 (new formula)

### PR DESCRIPTION
**Pull request by maijs**
[MailHog](https://github.com/mailhog/MailHog) is a Go-based email testing tool for developers.

This formula builds and installs MailHog version 0.1.8.

### CODE REVIEW

File: Library/Formula/mailhog.rb [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+      <string>#{var}/log/mailhog.log</string>
+      <key>StandardOutPath</key>
+      <string>#{var}/log/mailhog.log</string>
+    </dict>
+    </plist>
+    EOS
+  end
+
+  test do
+    output = shell_output("#{bin}/MailHog -h 2>&1 | head -n 1")
```
Comment (bfontaine) [inline Library/Formula/mailhog.rb]: Can’t we write a better test than e.g. `-h`/`-help`/`-version`?

File: Library/Formula/mailhog.rb [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+    bin.install "MailHog"
+  end
+
+  def caveats; <<-EOS.undent
+    To use MailHog as a sendmail replacement for PHP, add the
+    following line to php.ini file:
+
+      sendmail_path = #{HOMEBREW_PREFIX}/bin/MailHog sendmail test@test
+    EOS
+  end
```
Comment (bfontaine) [inline Library/Formula/mailhog.rb]: We shouldn’t include this unless this is Homebrew-specific.

File: Library/Formula/mailhog.rb (hunk 1 above)
Comment (maijs) [inline Library/Formula/mailhog.rb]: Valid point. Given that `MailHog` outputs debug messages when run directly, test should check against that.

File: Library/Formula/mailhog.rb (hunk 2 above)
Comment (maijs) [inline Library/Formula/mailhog.rb]: It seems that there's no `caveats` equivalent for info messages, will use regular `puts` in `def install` to inform users about sendmail replacement option.

File: Library/Formula/mailhog.rb [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+
+    ENV["GOPATH"] = buildpath
+    Language::Go.stage_deps resources, buildpath/"src"
+    system "go", "build", "-o", "MailHog"
+    bin.install "MailHog"
+
+    puts <<-EOS.undent
+    To use MailHog as a sendmail replacement for PHP, add the following line to php.ini file:
+      sendmail_path = #{HOMEBREW_PREFIX}/bin/MailHog sendmail test@test
+    EOS
```
Comment (bfontaine) [inline Library/Formula/mailhog.rb]: Please don’t print this. We use caveats for Homebrew- or OSX-specific message; this is a general message that should be in the software documentation instead.

Comment (maijs) [inline Library/Formula/mailhog.rb]: MailHog project page [has information](https://github.com/mailhog/MailHog#sendmail) about sendmail replacement, hence message removal from `def install`.

File: Library/Formula/mailhog.rb [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+    <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+    <plist version="1.0">
+    <dict>
+      <key>KeepAlive</key>
+      <true/>
+      <key>Label</key>
+      <string>#{plist_name}</string>
+      <key>ProgramArguments</key>
+      <array>
+        <string>#{HOMEBREW_PREFIX}/bin/MailHog</string>
```
Comment (DomT4) [inline Library/Formula/mailhog.rb]: This should be `opt_bin` rather than `#{HOMEBREW_PREFIX}/bin/`

Comment (maijs) [inline Library/Formula/mailhog.rb]: @DomT4 Thanks, noted.

**Comment by maijs** [comment]
Is there anything else that needs to be done for this formula to be merged?

File: Library/Formula/mailhog.rb [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+      :revision => "8fd9a4b75098b2125fe442e48a3ffbf738254e13"
+  end
+
+  go_resource "labix.org/v2/mgo" do
+    url "https://github.com/go-mgo/mgo.git",
+      :revision => "2e26580ebcca4eca6533535f2dd062cd6bff44ac"
+  end
+
+  def install
+    mkdir_p "#{buildpath}/src/github.com/mailhog/"
```
Comment (bfontaine) [inline Library/Formula/mailhog.rb]: Use `(buildpath/"src/github.com/mailhog/").mkpath`

File: Library/Formula/mailhog.rb [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+      <key>StandardErrorPath</key>
+      <string>#{var}/log/mailhog.log</string>
+      <key>StandardOutPath</key>
+      <string>#{var}/log/mailhog.log</string>
+    </dict>
+    </plist>
+    EOS
+  end
+
+  test do
```
Comment (DomT4) [inline Library/Formula/mailhog.rb]: Can we use the [mpd](https://github.com/Homebrew/homebrew/blob/master/Library/Formula/mpd.rb#L129-L141) style test here instead?

Comment (maijs) [inline Library/Formula/mailhog.rb]: @DomT4 Indeed, utilizing `Process.kill` is a leaner approach.

**Comment by DomT4** [comment]
Merged in https://github.com/Homebrew/homebrew/commit/4e4cab4556bcf1e43dd9fcca511bf0536351bcb2. Thank you for your contribution to Homebrew @maijs; we appreciate it! :smiley_cat:

**Pull request closed without merging**